### PR TITLE
Add data source online updates to the Update 4 MII sample.

### PR DESCRIPTION
**Pull request by tbarnes-us**
This adds datasource updates into MII sample use case Update 4.

### CODE REVIEW

File: docs-source/content/samples/simple/domains/model-in-image/_index.md [hunk 1]
```diff
@@ -14,14 +14,15 @@ description: "Sample for supplying a WebLogic Deploy Tooling (WDT) model that th
...
-   - [Update 4]({{< relref "/samples/simple/domains/model-in-image/update4.md" >}}): Dynamically update Work Manager Threads Constraints configurations using a model ConfigMap without restarting servers
-   - [Cleanup]({{< relref "/samples/simple/domains/model-in-image/cleanup.md" >}})
+   - Sample steps
+     - [Prerequisites for all domain types]({{< relref "/samples/simple/domains/model-in-image/prerequisites#prerequisites-for-all-domain-types" >}})
+     - [Additional prerequisites for JRF domains]({{< relref "/samples/simple/domains/model-in-image/prerequisites#additional-prerequisites-for-jrf-domains" >}})
+     - [Initial]({{< relref "/samples/simple/domains/model-in-image/initial.md" >}}) use case: An initial WebLogic domain
+     - [Update 1]({{< relref "/samples/simple/domains/model-in-image/update1.md" >}}): Dynamically adding a data source using a model ConfigMap and a domain restart (roll)
+     - [Update 2]({{< relref "/samples/simple/domains/model-in-image/update2.md" >}}): Deploying an additional domain
+     - [Update 3]({{< relref "/samples/simple/domains/model-in-image/update3.md" >}}): Updating an application in an image
```
Comment (alai8) [inline docs-source/content/samples/simple/domains/model-in-image/_index.md:23]: Should we add "and a domain restart (roll)" for update 3 as in update 1?

File: kubernetes/samples/scripts/create-oracle-db-service/common/oracle.db.yaml [hunk 2]
```diff
@@ -71,4 +71,4 @@ spec:
       securityContext: {}
```
Comment (tbarnes-us) [inline kubernetes/samples/scripts/create-oracle-db-service/common/oracle.db.yaml:71]: I checked this file in accidentally and will remove it.  It's implicitly updated when you start the DB using the DB sample...

File: kubernetes/samples/scripts/create-weblogic-domain/model-in-image/utils/wl-pod-wait.sh [hunk 3]
```diff
@@ -172,6 +176,32 @@ function print_table() {
...
+# and place result in the env var named by $2
+# if expected>0 and echo an Error and exit script non-zero
+#
+function getDomainValue() {
+  local attvalue
+  local ljpath="{$1}"
+  local __retvar=$2
+  set +e
+  attvalue=$(kubectl -n ${DOMAIN_NAMESPACE} get domain ${DOMAIN_UID} -o=jsonpath="$ljpath" 2>&1)
+  # attvalue=$(kubectl -n ${DOMAIN_NAMESPACE} get domain ${DOMAIN_UID} -o=jpath='{.spec.restartVersion}' 2>&1)
```
Comment (alai8) [inline kubernetes/samples/scripts/create-weblogic-domain/model-in-image/utils/wl-pod-wait.sh:191]: should this commented line be removed?

File: src/integration-tests/model-in-image/mii-sample-wrapper/README [hunk 4]
```diff
@@ -78,8 +78,20 @@ INCLUDE_MODEL_CONFIGMAP ::: Tell sample to include a configuration.model.configM
...
+  The DS secret is used by the configMap's DS to point to a DB.
+  It's deliberately incorrect by default so the sample
+  can demonstrate dynamically updating the DS params using online.
+  See also INCLUDE_MODEL_CONFIGMAP.
+  Valid values are 'false' (default), and 'true'.
+
+http_proxy https_proxy ::: Proxies
+  If running behind a proxy, then set as needed to allow curl access
+  to github.com. Used by 'stage-tooling.sh'.
+
```
Comment (tbarnes-us) [inline src/integration-tests/model-in-image/mii-sample-wrapper/README:94]: Looks like I doubled the http_proxy entry - I will fix.

File: src/integration-tests/model-in-image/mii-sample-wrapper/README [hunk 5]
```diff
@@ -78,8 +78,20 @@ INCLUDE_MODEL_CONFIGMAP ::: Tell sample to include a configuration.model.configM
...
   Valid values are 'false' (default), and 'true'.
 
+CORRECTED_DATASOURCE_SECRET ::: Use correct values in datasource secret
+  The DS secret is used by the configMap's DS to point to a DB.
+  It's deliberately incorrect by default so the sample
+  can demonstrate dynamically updating the DS params using online.
+  See also INCLUDE_MODEL_CONFIGMAP.
+  Valid values are 'false' (default), and 'true'.
+
+http_proxy https_proxy ::: Proxies
```
Comment (alai8) [inline src/integration-tests/model-in-image/mii-sample-wrapper/README:91]: duplicate

File: src/integration-tests/model-in-image/mii-sample-wrapper/generate-sample-doc.sh [hunk 6]
```diff
@@ -132,6 +137,9 @@ for phase in initial update1 update2 update3 update4; do
   if [ $configmap != "None" ]; then
     export INCLUDE_MODEL_CONFIGMAP=true
   fi
+  if [ $corrected_datasource_secret = "true" ]; then
+    export CORRECTED_DATASOURCE_SECRET=true
```
Comment (tbarnes-us) [inline src/integration-tests/model-in-image/mii-sample-wrapper/generate-sample-doc.sh:141]:
```
if [ $corrected_datasource_secret = "true" ]; then
    export CORRECTED_DATASOURCE_SECRET=true
fi
```
should be replaced with
```
export CORRECTED_DATASOURCE_SECRET=$corrected_datasource_secret
```

Comment (alai8) [reply src/integration-tests/model-in-image/mii-sample-wrapper/generate-sample-doc.sh:141]: yes, this is better, but what it was before would also work

File: docs-source/content/samples/simple/domains/model-in-image/update4.md [hunk 7]
```diff
@@ -54,6 +57,31 @@ Here are the steps:
...
+1. Optionally start the database.
+
+    - If the database is running, then the sample application that we will run at the end of this step will verify that your updates to the data source secret took effect.
+
+    - If you are taking the `JRF` path through the sample, then the database will already be running. 
+
+    - If you are taking the `WLS` path through the sample, then you can deploy the database by:
+      - Following the the first step in [Set up and initialize an infrastructure database]({{< relref "/samples/simple/domains/model-in-image/prerequisites.md#set-up-and-initialize-an-infrastructure-database" >}}). This step is titled `Ensure that you have access to the database image, and then create a deployment`.
+      - You can skip the remaining steps (they are only needed for `JRF`).
+
```
Comment (tbarnes-us) [inline docs-source/content/samples/simple/domains/model-in-image/update4.md:84]: the kubectl edit command could use a little more help - provide the following snippet (substitute JRF for WLS for the JRF path):
```
  configuration:
    model:
      configMap: sample-domain1-wdt-config-map
      domainType: WLS
      onlineUpdate:
        enabled: true
      runtimeEncryptionSecret: sample-domain1-runtime-encryption-secret
```

Comment (tbarnes-us) [reply docs-source/content/samples/simple/domains/model-in-image/update4.md:84]: Also, the patch command should use 'true' not "true".

File: docs-source/content/samples/simple/domains/model-in-image/_index.md (hunk 1 above)
Comment (tbarnes-us) [reply docs-source/content/samples/simple/domains/model-in-image/_index.md]: Sure. Done.

File: kubernetes/samples/scripts/create-weblogic-domain/model-in-image/utils/wl-pod-wait.sh (hunk 3 above)
Comment (tbarnes-us) [reply kubernetes/samples/scripts/create-weblogic-domain/model-in-image/utils/wl-pod-wait.sh]: Yes. Done.

File: src/integration-tests/model-in-image/mii-sample-wrapper/README (hunk 5 above)
Comment (tbarnes-us) [reply src/integration-tests/model-in-image/mii-sample-wrapper/README:91]: Unduplicated. Done.

**Pull request merged**